TypeCobol/Compiler/CodeElements/Paragraph/RepositoryParagraph.cs

```diff
@@ -1,21 +1,46 @@
 ﻿using System;
 using System.Collections.Generic;
+using TypeCobol.Compiler.Scanner;
 
 namespace TypeCobol.Compiler.CodeElements
 {
     /// <summary>
     /// The REPOSITORY paragraph is used in a program or class definition to identify all
     /// the object-oriented classes that are intended to be referenced in that program or
-    /// class definition.
+    /// class definition. It also allows declaration of intrinsic function names that may be used without
+    /// specifying the keyword FUNCTION.
     /// </summary>
     public class RepositoryParagraph : CodeElement
     {
+        /// <summary>
+        /// Intrinsic functions which are not allowed in the REPOSITORY paragraph
+        /// </summary>
+        public static readonly HashSet<string> NotAllowedIntrinsicFunctions = new[] { "WHEN-COMPILED" }.ToHashSet(StringComparer.OrdinalIgnoreCase);
+
         public RepositoryParagraph() : base(CodeElementType.RepositoryParagraph)
         { }
 
         /// <summary>
         /// Optionally, the REPOSITORY paragraph defines associations between class-names and external class-names.
         /// </summary>
         public IDictionary<SymbolDefinitionOrReference, SymbolDefinitionOrReference> ClassNames { get; set; }
+
+        /// <summary>
+        /// Optionally, the REPOSITORY paragraph allows to declare all intrinsic functions
+        /// </summary>
+        public bool IsAllIntrinsicFunctions { get; set; }
+
+        /// <summary>
+        /// Optionally, the REPOSITORY paragraph allows to declare one or several intrinsic functions
+        /// </summary>
+        public List<string> IntrinsicFunctions { get; set; }
```

Comment:
Use `List<SymbolDefinitionOrReference>` to capture tokens as well

Reply:
Done